src/material.rs

```diff
@@ -0,0 +1,45 @@
+use render::BasicPipelineState;
+use scene::Color;
+use texture::Texture;
+
+/// Material is the enhancement of Texture that is used to setup appearance of [`Mesh`](struct.Mesh.html).
+#[allow(missing_docs)]
+#[derive(Clone, Debug)]
+pub enum Material {
+    /// Basic wireframe with specific `Color`.
+    LineBasic { color: Color },
+    /// Basic material with color, optional `Texture` and optional wireframe mode.
+    MeshBasic {
+        color: Color,
+        map: Option<Texture<[f32; 4]>>,
+        wireframe: bool,
+    },
+    /// Lambertian diffuse reflection. This technique causes all closed polygons
+    /// (such as a triangle within a 3D mesh) to reflect light equally in all
+    /// directions when rendered.
+    MeshLambert { color: Color, flat: bool },
+    /// Material that uses Phong reflection model.
+    MeshPhong { color: Color, glossiness: f32 },
+    /// Physically-based rendering material.
+    MeshPbr {
```

Comment:
three.js names their PBR material type `MeshStandardMaterial`. I used `Pbr` because it's shorter and more descriptive. Is this OK?

Reply:
I think it's okay. `MeshStandardMaterial` is too verbose (`Material::MeshStandardMaterial`)

Reply:
Moreover, we have `MeshBasic`. "Basic" and "Standard" - what to choose, huh?

Reply:
I agree.